src/modules/game-client/game-client.service.ts

```diff
@@ -55,79 +56,95 @@ export class GameClientService {
   async handleBuzzerRequest(socketId: string) {
     const gameId = await this.gameStateRepository.getGameIdBySocketId(socketId);
     const gameState =
-      await this.gameStateRepository.getGameState<GameStatus.ROUND_IN_PROGRESS>(
-        gameId,
-      );
+        await this.gameStateRepository.getGameState<GameStatus.ROUND_IN_PROGRESS>(
+          gameId,
+        );
+    const player: Player = gameState.gamePlayers[socketId];
 
-    const player = gameState.gamePlayers[socketId];
+    if (socketId === gameState.roundData.buzzersGranted[gameState.roundData.buzzersGranted.length]){
```

Comment:
why are you checking if this is a buzzer twice here?

shouldn’t you also check if its allowed in the settings

Reply:
didn’t understand this logic brother